### PR TITLE
Simulator initial draft

### PR DESCRIPTION
**Pull request by 0x4d**
## Changes
- Added SimulationManagerModule to keep track of available simulators and simulation state
- Added MetaSimulationModule to allow simulation through MetaFormalismModule
- Added ExternalSimulationModule to allow simulation through external simulators using [renew-simulator-gateway](https://github.com/renew-js/renew-simulator-gateway)

## TODO
- Add tests

### CODE REVIEW

File: src/features/external-simulation/providers/ExternalSimulation.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        this.simulationManager = simulationManager;
+        this.simulators = [];
+        this.socket = io('http://localhost:3000/', {
+            'path': '/gateway',
+        });
+        this.registerHandlers();
+    }
+
+    registerHandlers () {
+        this.socket.on('connect', () => {
```
Comment (tim-kilian) [inline src/features/external-simulation/providers/ExternalSimulation.js]: Why is this not in renew-simulator-client?

Comment (0x4d) [reply src/features/external-simulation/providers/ExternalSimulation.js]: I initially discarded the renew-simulator-client and decided to use the socket directly in the external-simulation module. Another abstraction might be useful however for other people planning JavaScript projects using the renew-simulator-gateway. I'll think about it.

**Comment by tim-kilian** [comment]
Set to WIP, util the CI is not green. The merge is missing.

**Comment by 0x4d** [comment]
The PR build passed now, after I removed the cache and restarted it.

**Comment by tim-kilian** [comment]
Cannot merge it automatically

**Review by tim-kilian — APPROVED** [review]
Looks solid

**Pull request merged**